File: julia/abc200_209/abc201_C.jl

using Random
infile = stdin
## Type Shortcuts (to save my wrists and fingers :))
const I = Int64; const VI = Vector{I}; const SI = Set{I}; const PI = NTuple{2,I};
const TI = NTuple{3,I}; const QI = NTuple{4,I}; const VPI = Vector{PI}; const SPI = Set{PI}
const VC = Vector{Char}; const VS = Vector{String}; VB = Vector{Bool}; VVI = Vector{Vector{Int64}}
const F = Float64; const VF = Vector{F}; const PF = NTuple{2,F}

gs()::String = rstrip(readline(infile))
gi()::Int64 = parse(Int64, gs())
gf()::Float64 = parse(Float64,gs())
gss()::Vector{String} = split(gs())
gis()::Vector{Int64} = [parse(Int64,x) for x in gss()]
gfs()::Vector{Float64} = [parse(Float64,x) for x in gss()]

function solve(S::String)
    ans::I = 0
    for i1 in 0:9; for i2 in 0:9; for i3 in 0:9; for i4 in 0:9
        s = "$i1$i2$i3$i4"
        good::Bool = true
        for i in 0:9
            if S[i+1] == 'o' && '0'+i ∉ s; good = false; break; end
            if S[i+1] == 'x' && '0'+i ∈ s; good = false; break; end
        end
        if good; ans += 1; end
    end; end; end; end
    return ans
end

function main(infn="")
    global infile
    infile = (infn != "") ? open(infn,"r") : length(ARGS) > 0 ? open(ARGS[1],"r") : stdin
    S::String = gs()
    ans = solve(S)
    println(ans)
end

Random.seed!(8675309)
main()

#using Profile, StatProfilerHTML
#Profile.clear()
#@profile main("B.in")
#Profile.clear()
#@profilehtml main("B.in")

File: python/abc200_209/abc201_C.py

import sys
infile = sys.stdin.buffer

def gs()  : return infile.readline().rstrip()
def gi()  : return int(gs())
def gf()  : return float(gs())
def gss() : return gs().split()
def gis() : return [int(x) for x in gss()]
def gfs() : return [float(x) for x in gss()]

def main(infn="") :
    global infile
    infile = open(infn,"r") if infn else open(sys.argv[1],"r") if len(sys.argv) > 1 else sys.stdin
    S = gs()
    ans = 0
    for pin in range(10000) :
        a = pin // 1000; pin -= 1000*a
        b = pin // 100; pin -= 100*b
        c = pin // 10; pin -= 10*c
        d = pin
        good = True
        for i in range(10) :
            if S[i] == 'o' and i not in (a,b,c,d) : good = False; break
            if S[i] == 'x' and i in (a,b,c,d) : good = False; break
        if good : ans += 1
    sys.stdout.write(str(ans)+'\n')

if __name__ == '__main__' :
    main()
    sys.stdout.flush()

